core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -233,6 +251,19 @@ public void perform(Run<?,?> build, FilePath ws, Launcher launcher, TaskListener
 
             Map<String,String> files = ws.act(new ListFiles(artifacts, excludes, defaultExcludes, caseSensitive));
             if (!files.isEmpty()) {
+                if (isFlattenDirectories()) {
+                    Map<String,String> flattenedFiles = new LinkedHashMap<>();
+                    for (Map.Entry<String,String> soloFile : files.entrySet()) {
+                        String simpleFile = soloFile.getKey().replaceFirst(".+/", "");
+                        if (flattenedFiles.put(simpleFile, soloFile.getKey()) != null) {
+                            throw new AbortException("Attempted to add duplicate filenames to flattened archive. " +
```

Comment:
I disagree with oleg :-)
Aborting is the right thing by default in the abscence of anything else.  Logs don;t get read and often by the time out find out someone else introduced a "updateDB.sh" its too late and your production server is messed up as it picked the wrong one and continued the rest of the pipeline...